packages/integration-tests/__helpers__/awaitUrlChange.js

```diff
@@ -21,3 +21,24 @@ const haschanged = async (currentUrl, tries = 1) => {
   if (currentUrl !== window.location.href) return
   return haschanged(currentUrl, ++tries)
 }
+
+export const windowHistoryGo = (n) =>
+  new Promise((resolve, reject) => {
+    let timeout
+
+    // Resolve when the resulting popstate event happens
+    const onPopState = () => {
+      clearTimeout(timeout)
+      resolve()
+    }
+    window.addEventListener('popstate', onPopState)
+
+    // Trigger the real go function
+    window.history.go(n)
+
+    // timeout after 1 second
+    timeout = setTimeout(() => {
```

Comment:
Need a clear timeout here otherwise memory leaks happen. I can send you a gist of it or google “settimeout memory leak”

Reply:
Hmm, as far as I can tell its not necessary to clear a timeout that has already happened - all the exampoes I can find of buggy code involve a `setInterval`, or code that does `setTimeout` again inside the `setTimeout` callback.

See https://stackoverflow.com/questions/4244309/is-there-memory-leak-issue-if-not-calling-cleartimout-after-calling-settimeout

I can't imagine why the memory can't be freed if the timeout has already happened.

Reply:
Following that logic, everytime `setTimeout` is called and the return value is ignored, there would be a memory leak - since without assigning the return value there is nothing to call `clearTimeout` on?

Reply:
You got me thinking though... there was at least one memory leak in this code because the `popstate` listener (only meant to be called once) was not removed the first time it was called. I fixed that up.